### PR TITLE
Scala3 support for jms connector

### PR DESCRIPTION
**Pull request by pjfanning**
Part of https://github.com/apache/incubator-pekko-connectors/issues/126

Scala 3 compile issues needed fixing.
* main ones were related to using protected methods on self types
* also some issues where fields were ambiguous (multiple inheritance in Scala means that a field can be declared on 2 separate traits)

I introduced a GraphStageCompanion trait to give public access to the methods that were needed to get Scala3 compile to work.

### CODE REVIEW

File: jms/src/main/scala/org/apache/pekko/stream/connectors/jms/impl/GraphStageCompanion.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * Exposes some protected methods from [[pekko.stream.stage.GraphStage]]
+ * that are not accessible when using Scala3 compiler.
+ */
+@InternalApi
+private[impl] trait GraphStageCompanion {
+  def graphStageMaterializer: Materializer
+
+  def graphStageDestination: Destination
+
+  def scheduleOnceOnGraphStage(timerKey: Any, delay: FiniteDuration): Unit
```
Comment (mdedetrich) [inline jms/src/main/scala/org/apache/pekko/stream/connectors/jms/impl/GraphStageCompanion.scala:29]: Can we avoid using `Any` here?

**Review by mdedetrich — CHANGES_REQUESTED** [review]
Some more cases of `Any`

File: jms/src/main/scala/org/apache/pekko/stream/connectors/jms/impl/GraphStageCompanion.scala (hunk 1 above)
Comment (pjfanning) [reply jms/src/main/scala/org/apache/pekko/stream/connectors/jms/impl/GraphStageCompanion.scala:29]: This signature is taken from the function I am delegating to.

Comment (pjfanning) [reply jms/src/main/scala/org/apache/pekko/stream/connectors/jms/impl/GraphStageCompanion.scala:29]: @mdedetrich is this answer ok? The timerKey can be any case class, the 'Any' type is actually what makes sense. This is also an internal class, not an API one.

Comment (mdedetrich) [reply jms/src/main/scala/org/apache/pekko/stream/connectors/jms/impl/GraphStageCompanion.scala:29]: Let me check this now, we really shouldn't have `Any` like this so something fishy is going on.

**Review by mdedetrich — APPROVED** [review]
Okay I had a deeper look and you are right, it also uses `Any` with the current version (i.e. `scheduleOnce`). This probably should be looked into but its not relevant to Scala 3

**Review by mdedetrich — APPROVED** [review]
lgtm

**Pull request merged**